plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/DiskCache.kt

```diff
@@ -162,11 +174,18 @@ class DiskCache(
     override fun loadAccessToken(cacheKey: AccessTokenCacheKey): AccessToken? {
         LOG.debug { "loadAccessToken for $cacheKey" }
         val cacheFile = accessTokenCache(cacheKey)
-        val inputStream = cacheFile.tryInputStreamIfExists() ?: return null
-
-        val token = loadAccessToken(inputStream)
-
-        return token
+        // If file exists, returns InputStream, if not returns null
+        return cacheFile.tryInputStreamIfExists()
+            //try to load and parse access token, returns AccessToken or null if expired
+            ?.let { loadAccessToken(it) }
+            // If file doesn't exist or loadAccessToken failed, try in-memory cache
```

Comment:
what happens if the state on disk does not match what is in memory?

Reply:
should check memory first I think for the load functions. Because if this is the case that there exists both a file and in-memory token(ClientReg too), then that means the most recent state that attempted to save would not fit on disk, so that should be checked first.